webapp/views/App/views/Home/Dashboard/ActivityLog/store/parsers/isItemDeletedFns/category.js

```diff
@@ -9,45 +11,45 @@ import * as ActivityLogMessageParserUtils from '../utils'
 export default {
   // ===== CATEGORY
 
-  [ActivityLog.type.categoryInsert]: ActivityLogMessageParserUtils.isCategoryDeleted,
+  [ActivityLog.type.categoryInsert]: () => ActivityLogMessageParserUtils.isCategoryDeleted,
 
-  [ActivityLog.type.categoryPropUpdate]: ActivityLogMessageParserUtils.isCategoryDeleted,
+  [ActivityLog.type.categoryPropUpdate]: () => ActivityLogMessageParserUtils.isCategoryDeleted,
 
   // ===== CATEGORY LEVEL
 
-  [ActivityLog.type.categoryLevelInsert]: (survey) => (activityLog) => {
+  [ActivityLog.type.categoryLevelInsert]: () => (activityLog) => {
     const categoryLevelInserted = ActivityLog.getContent(activityLog)
-    const category = ActivityLogMessageParserUtils.getItemCategory(survey)(activityLog)
+    const category = ActivityLogMessageParserUtils.getItemCategory(activityLog)
     const levelUuid = CategoryLevel.getUuid(categoryLevelInserted)
 
-    return !category || !Category.getLevelByUuid(levelUuid)(category)
+    return A.isEmpty(category) || !Category.getLevelByUuid(levelUuid)(category)
   },
 
-  [ActivityLog.type.categoryLevelPropUpdate]: (survey) => (activityLog) => {
-    const category = ActivityLogMessageParserUtils.getItemCategory(survey)(activityLog)
+  [ActivityLog.type.categoryLevelPropUpdate]: () => (activityLog) => {
+    const category = ActivityLogMessageParserUtils.getItemCategory(activityLog)
     const levelUuid = ActivityLog.getContentUuid(activityLog)
 
-    return !category || !Category.getLevelByUuid(levelUuid)(category)
+    return A.isEmpty(category) || !Category.getLevelByUuid(levelUuid)(category)
   },
 
-  [ActivityLog.type.categoryLevelDelete]: ActivityLogMessageParserUtils.isItemCategoryDeleted,
+  [ActivityLog.type.categoryLevelDelete]: () => ActivityLogMessageParserUtils.isItemCategoryDeleted,
 
   // ===== CATEGORY ITEM
 
-  [ActivityLog.type.categoryItemInsert]: (survey) => (activityLog) => {
-    const category = ActivityLogMessageParserUtils.getItemCategory(survey)(activityLog)
+  [ActivityLog.type.categoryItemInsert]: () => (activityLog) => {
+    const category = ActivityLogMessageParserUtils.getItemCategory(activityLog)
     const itemInserted = ActivityLog.getContent(activityLog)
     const levelUuid = CategoryItem.getLevelUuid(itemInserted)
 
-    return !category || !Category.getLevelByUuid(levelUuid)(category)
+    return A.isEmpty(category) || !Category.getLevelByUuid(levelUuid)(category)
   },
 
-  [ActivityLog.type.categoryItemPropUpdate]: (survey) => (activityLog) => {
-    const category = ActivityLogMessageParserUtils.getItemCategory(survey)(activityLog)
+  [ActivityLog.type.categoryItemPropUpdate]: () => (activityLog) => {
+    const category = ActivityLogMessageParserUtils.getItemCategory(activityLog)
     const levelUuid = ActivityLog.getContentLevelUuid(activityLog)
 
-    return !category || !Category.getLevelByUuid(levelUuid)(category)
+    return A.isEmpty(category) || !Category.getLevelByUuid(levelUuid)(category)
```

Comment:
not sure about the logic that is there before, after confirm that I will extract this conditional above because is repeated 3 times

Reply:
that checks that the level still exists. yes, better to have one method that does that.
```const isLevelDeleted = ({category, levelUuid }) => A.isEmpty(category) || !Category.getLevelByUuid(levelUuid)(category)```

Reply:
please add the common method